src/utils/download.utils.ts

```diff
@@ -87,15 +111,48 @@ function downloadResource<T>(
     request.end();
   });
 }
-function getPortNumber(
-  parsedUrl: url.UrlWithStringQuery
-): string | number | undefined {
-  return (
-    (parsedUrl.port && Number(parsedUrl.port)) ||
-    (parsedUrl.protocol === "https:" ? 443 : 80)
-  );
+
+export function getHttpStatusCode(urlStr: string): Promise<number | undefined> {
+  return new Promise<number | undefined>((resolve, reject) => {
+    const parsedUrl = new URL(urlStr);
+    https
+      .request(
+        {
+          protocol: parsedUrl.protocol,
+          hostname: parsedUrl.hostname,
+          port: parsedUrl.port,
+          pathname: parsedUrl.pathname,
+          path: parsedUrl.pathname,
+          rejectUnauthorized: !tabnineExtensionProperties.ignoreCertificateErrors,
+          ca: tabnineExtensionProperties.caCerts
+            ? readCaCertsSync(tabnineExtensionProperties.caCerts)
+            : undefined,
+          agent: getHttpAgent(),
+          headers: { "User-Agent": "TabNine.tabnine-vscode" },
+          timeout: 30_000,
+        },
+        (response) => {
+          resolve(response.statusCode);
+          response.on("error", (error) => {
+            reject(error);
+          });
+          return undefined;
+        }
+      )
+      .on("error", (error) => {
+        reject(error);
+      });
+  });
 }
 
-function getHttpClient(parsedUrl: url.UrlWithStringQuery) {
-  return parsedUrl.protocol === "https:" ? https : http;
+export function readCaCertsSync(caCertsFileName: string): Buffer | undefined {
+  try {
+    if (!caCertsFileName) {
+      return undefined;
+    }
+    return fs.readFileSync(caCertsFileName);
+  } catch (error) {
+    Logger.warn("Failed to read CA certs file", error);
```

Comment:
For now we'll only see it in the logs, then?

Reply:
yes.. this is not exposed for now